### PR TITLE
feat: add GPU metrics support to CLI

### PR DESCRIPTION
**Pull request by munishchouhan**
## Summary                                                                                                                                                                       
   
  - Add `gpu` metric type to `tw runs view metrics` command showing aggregate `gpuUsage`, `gpuMemPeak`, `gpuMemAvg` stats per process (included by default or via `-t gpu`)        
  - Add GPU metrics to `tw runs view task --resources-usage` output showing per-task GPU details (name, driver, utilisation, memory, bandwidth, active time)                     
  - Add `formatMemoryMiB` helper for formatting GPU memory values                                                                                                                  
  - Update GraalVM reflection config for `GpuMetrics` native image support                                                                                                         
                                                                                                                                                                                   
  ## Usage                                                                                                                                                                         
                                                                                                                                                                                   
  ```bash                                                                                                                                                                        
  # Aggregate GPU metrics per process
  tw runs view -i <runId> metrics -t gpu
                                                                                                                                                                                   
  # Per-task GPU details (included in --resources-usage)
  tw runs view -i <runId> task -t <taskId> --resources-usage                                                                                                                       
                                                                                                                                                                                   
  # JSON output
  tw runs view -i <runId> task -t <taskId> --resources-usage -o json                                                                                                               
                                                                                                                                                                                   
  Test plan
                                                                                                                                                                                   
  - Existing MetricsCmdTest and TaskCmdTest pass                                                                                                                                   
  - Full test suite passes
  - Manual test: tw runs view task -t 1 --resources-usage shows GPU metrics (name, driver, utilisation, memory, active time)                                                       
  - Manual test: tw runs view metrics -t gpu works                                                                                                                                 
  - Native binary (nativeCompile) works with GPU metrics deserialization

### CODE REVIEW

**Comment by munishchouhan** [comment]
tested locally:
```
tower-cli % ./build/native/nativeCompile/tw runs view -i 3mBg2gAJrS9JLA -w SeqeraLabs-UnifiedCompute/munish metrics -t gpu


     GPU Metrics
    ----------------

     Legend: utilisation / peak memory / avg memory

     process      | min                   | q1                    | q2                    | q3                    | max                   | mean                  
    --------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------
     GPU_CONV     |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB 
     GPU_MATMUL   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   
     GPU_TRAINING |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   
 
tower-cli % ./build/native/nativeCompile/tw runs view -i 3mBg2gAJrS9JLA -w SeqeraLabs-UnifiedCompute/munish metrics       


     Memory Metrics
    ----------------

     Legend:  physical RAM / virtual RAM+swap / %RAM allocated 

     process      | min                | q1                 | q2                 | q3                 | max                | mean               
    --------------+--------------------+--------------------+--------------------+--------------------+--------------------+--------------------
     GPU_CONV     |  620MB   23GB   4% |  620MB   23GB   4% |  620MB   23GB   4% |  620MB   23GB   4% |  620MB   23GB   4% |  620MB   23GB   4% 
     GPU_MATMUL   |  483MB    8GB   3% |  483MB    8GB   3% |  483MB    8GB   3% |  483MB    8GB   3% |  483MB    8GB   3% |  483MB    8GB   3% 
     GPU_TRAINING |  683MB    9GB   4% |  683MB    9GB   4% |  683MB    9GB   4% |  683MB    9GB   4% |  683MB    9GB   4% |  683MB    9GB   4% 


     CPU Metrics
    ----------------

     Legend: raw usage / % allocated

     process      | min         | q1          | q2          | q3          | max         | mean        
    --------------+-------------+-------------+-------------+-------------+-------------+-------------
     GPU_CONV     |   99%   25% |   99%   25% |   99%   25% |   99%   25% |   99%   25% |   99%   25% 
     GPU_MATMUL   |  100%   25% |  100%   25% |  100%   25% |  100%   25% |  100%   25% |  100%   25% 
     GPU_TRAINING |  100%   25% |  100%   25% |  100%   25% |  100%   25% |  100%   25% |  100%   25% 


     Time Metrics
    ----------------

     Legend: execution real-time / % requested time used

     process      | min      | q1       | q2       | q3       | max      | mean     
    --------------+----------+----------+----------+----------+----------+----------
     GPU_CONV     |    1m32s |    1m32s |    1m32s |    1m32s |    1m32s |    1m32s 
     GPU_MATMUL   |     2m2s |     2m2s |     2m2s |     2m2s |     2m2s |     2m2s 
     GPU_TRAINING |    1m33s |    1m33s |    1m33s |    1m33s |    1m33s |    1m33s 


     I/O Metrics
    ----------------

     Legend: reads / writes

     process      | min           | q1            | q2            | q3            | max           | mean          
    --------------+---------------+---------------+---------------+---------------+---------------+---------------
     GPU_CONV     |   13MB   752B |   13MB   752B |   13MB   752B |   13MB   752B |   13MB   752B |   13MB   752B 
     GPU_MATMUL   |   13MB   822B |   13MB   822B |   13MB   822B |   13MB   822B |   13MB   822B |   13MB   822B 
     GPU_TRAINING |   25MB   775B |   25MB   775B |   25MB   775B |   25MB   775B |   25MB   775B |   25MB   775B 


     GPU Metrics
    ----------------

     Legend: utilisation / peak memory / avg memory

     process      | min                   | q1                    | q2                    | q3                    | max                   | mean                  
    --------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------
     GPU_CONV     |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB |   88% 14.2 GB 12.6 GB 
     GPU_MATMUL   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   |   93% 390 MB 365 MB   
     GPU_TRAINING |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   |   88% 344 MB 309 MB   

```

**Comment by munishchouhan** [comment]
tested locally:
```
tower-cli % ./build/native/nativeCompile/tw runs view -i 3mBg2gAJrS9JLA -w SeqeraLabs-UnifiedCompute/munish task -t 1 --resources-usage


     General
    ----------------+--------------------------------------------------
     Task ID        | 1                                                
     ID             | 1300752644                                       
     Work directory | s3://nf-munish/54/9e2f45dab86a1600c516311ecc0566 
     Status         | COMPLETED                                        

     Command
    ---------
     python3 -c "
     import torch
     import time
     
     assert torch.cuda.is_available(), 'CUDA NOT AVAILABLE'
     device = torch.device('cuda')
     print(f'GPU: {torch.cuda.get_device_name(0)}')
     
     print('Running convolution benchmark...')
     model = torch.nn.Sequential(
         torch.nn.Conv2d(3, 64, 3, padding=1),
         torch.nn.ReLU(),
         torch.nn.Conv2d(64, 128, 3, padding=1),
         torch.nn.ReLU(),
         torch.nn.Conv2d(128, 256, 3, padding=1),
     ).to(device)
     
     start = time.time()
     while time.time() - start < 90:
         x = torch.randn(32, 3, 224, 224, device=device)
         y = model(x)
         torch.cuda.synchronize()
     
     print(f'Done in {time.time()-start:.0f}s')
     print(f'Peak GPU memory: {torch.cuda.max_memory_allocated()/1e9:.2f} GB')
     " | tee gpu_conv.txt


     Resources usage
    ------------------+------------+--------------------------------------------------------------------------------------------------------
     pcpu             | 99%        | Percentage of CPU used by the process                                                                  
     rss              | 620MB      | Real memory (resident set) size of the process                                                         
     peakRss          | 620MB      | Peak of real memory                                                                                    
     vmem             | 23GB       | Virtual memory size of the process                                                                     
     peakVmem         | 23GB       | Peak of virtual memory                                                                                 
     rchar            | 13MB       | Number of bytes the process read, using any read-like system call from files, pipes, tty, etc          
     wchar            | 752B       | Number of bytes the process wrote, using any write-like system call                                    
     readBytes        | 171MB      | Number of bytes the process directly read from disk                                                    
     writeBytes       | 0B         | Number of bytes the process originally dirtied in the page-cache (assuming they will go to disk later) 
     syscr            | 2KB        | Number of read-like system call invocations that the process performed                                 
     syscw            | 39B        | Number of write-like system call invocations that the process performed                                
     volCtxts         | 1310.0     | Number of voluntary context switches                                                                   
     invCtxt          | 297.0      | Number of involuntary context switches                                                                 
     gpuName          | Tesla T4   | GPU model name                                                                                         
     gpuDriver        | 580.126.09 | NVIDIA driver version                                                                                  
     gpuTotalMemory   | 15.0 GB    | Total GPU memory                                                                                       
     gpuPct           | 88%        | Average GPU compute utilisation                                                                        
     gpuPeak          | 100%       | Peak GPU compute utilisation                                                                           
     gpuPctMem        | 84%        | Average GPU memory utilisation                                                                         
     gpuPeakMem       | 95%        | Peak GPU memory utilisation                                                                            
     gpuAvgMem        | 12.6 GB    | Average GPU memory used                                                                                
     gpuPeakMemUsed   | 14.2 GB    | Peak GPU memory used                                                                                   
     gpuAvgMemBwUtil  | 88%        | Average memory bandwidth utilisation                                                                   
     gpuPeakMemBwUtil | 100%       | Peak memory bandwidth utilisation                                                                      
     gpuActiveTime    | 1m30s      | Time the process was actively using GPU   
```

**Review by jonmarti — APPROVED** [review]
LGTM

**Review by stefanoboriero — APPROVED** [review]
